packages/paste-core/components/checkbox/src/Checkbox.tsx

```diff
@@ -124,12 +148,12 @@ const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
       >
         <HiddenCheckbox
           {...safelySpreadBoxProps(props)}
-          checked={checked}
+          checked={mergedChecked}
           disabled={disabled}
           name={name}
           onChange={handleChange}
           aria-describedby={helpTextId}
-          aria-checked={indeterminate ? 'mixed' : checked}
+          aria-checked={indeterminate && 'mixed'}
```

Comment:
Careful here: there are 3 states for `aria-checked`. `null` | `"true"` | `"mixed"`

Reply:
Oops! Yah fixed it.